tests/testTriggers.go

```diff
@@ -6458,6 +6459,125 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
 	updateMetrics(*event)
 }
 
+//TriggerPowerOffVMs
+func TriggerPowerOffAllVMs(contexts *[]*scheduler.Context, recordChan *chan *EventRecord){
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(PowerOffAllVMs)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: PowerOffAllVMs,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	setMetrics(*event)
+	stepLog := "Power off all worker nodes test"
+	Step(stepLog, func() {
+		log.Infof(stepLog)
+		workerNodes:= node.GetWorkerNodes()
+		log.Infof("All worker nodes : %v", workerNodes)
+		var numberOfThread int = 5
+		var numberOfNodePerThread int
+		if len(workerNodes) < numberOfThread{
+			numberOfThread = len(workerNodes)
+			numberOfNodePerThread = 1
+		}else{
+			numberOfNodePerThread = len(workerNodes)/numberOfThread
+		}
+		var counter int = 0
+		nodesInThread := make([][]node.Node, numberOfThread)
+		for t := 0; t < numberOfThread;  t++ {
+			nodesInThread[t] = make([]node.Node, numberOfNodePerThread)
+			for n:=0; n < numberOfNodePerThread; n++{
+					nodesInThread[t][n] = workerNodes[counter]
```

Comment:
Need to check if `counter` does not go out of bounds, else we will get index our of range error here.

Reply:
Yes. I checked with different values